internal/integration/base/api.go

```diff
@@ -480,6 +482,63 @@ func (apiSuite *APISuite) UserDisks(ctx context.Context, node string, sizeGreate
 	return disks, nil
 }
 
+// AssertServicesRunning verifies that services are running on the node.
+func (apiSuite *APISuite) AssertServicesRunning(ctx context.Context, node string, serviceStatus map[string]string) {
+	nodeCtx := client.WithNode(ctx, node)
+
+	for svc, state := range serviceStatus {
+		resp, err := apiSuite.Client.ServiceInfo(nodeCtx, svc)
+		apiSuite.Require().NoError(err)
+		apiSuite.Require().NotNil(resp, "expected service %s to be registered", svc)
+
+		for _, svcInfo := range resp {
+			apiSuite.Require().Equal(state, svcInfo.Service.State, "expected service %s to have state %s", svc, state)
+		}
+	}
+}
+
+// AssertExpectedModules verifies that expected kernel modules are loaded on the node.
+func (apiSuite *APISuite) AssertExpectedModules(ctx context.Context, node string, expectedModules map[string]string) {
+	nodeCtx := client.WithNode(ctx, node)
+
+	fileReader, err := apiSuite.Client.Read(nodeCtx, "/proc/modules")
+	defer func() {
+		err = fileReader.Close()
```

Comment:
```suggestion
		apiSuite.Require().NoError(fileReader.Close())
```

and it should be below line 509, as we shouldn't close if the `Read` call fails